### DISCUSSION TITLE
Error running ase interface

### DISCUSSION THREAD

**Discussion by marian-code** [deepmodeling/deepmd-kit#817] [Q&A]
Hi I am getting this error when trying to compute predictions for Ge crystal frames using dpdata.

```bash
Traceback (most recent call last):
File "/opt/miniconda3/lib/python3.8/site-packages/tensorflow/python/framework/importer.py", line 496, in _import_graph_def_internal
results = c_api.TF_GraphImportGraphDefWithResults(
tensorflow.python.framework.errors_impl.InvalidArgumentError: NodeDef mentions attr 'T' not in Op<name=DescrptSeA; signature=coord:double, type:int32, natoms:int32, box:double, mesh:int32, davg:double, dstd:double -> descrpt:double, descrpt_deriv:double, rij:double, nlist:int32; attr=rcut_a:float; attr=rcut_r:float; attr=rcut_r_smth:float; attr=sel_a:list(int); attr=sel_r:list(int)>; NodeDef: {{node load/DescrptSeA}}. (Check whether your GraphDef-interpreting binary is up to date with your GraphDef-generating binary.).
```

I am using deepmd kit version 1.3.3 with GPU acceleration and cuda10.1. This is a fresh installation.

The commands are just:

```bash
dp train control_file.yaml
dp freeze -d . -o model.pb
```

I have an similar instalation on another PC with deepmdkit v1.3.2 and repeating this procedure there works OK.

**Comment by njzjz** [comment]
Do you use model v1.3 in v1.2 program?

**Reply by marian-code** [reply]
No. But that was the first thing that came to my mind too, so I double checked but that is not the case. It is a conda installation. I activate the virtual environment, run training, freeze the model and then still in the same venv I try to get prediction with dpdata. I labeled the question a bit wrong, but the principle is the same for dpdata and ASE. The error occurs when the script is loading the model from file.

```python
import deepmd.DeepPot as DeepPot
DeepPot(model_file_str)
```
The whole traceback looks something like this:

```bash
File "/home/rynik/Software/dpmd_tools/dpmd_tools/system/flavours/masked_system.py", line 366, in predict
self.dp = DeepPot(model)
File "/opt/miniconda3/lib/python3.8/site-packages/deepmd/DeepPot.py", line 14, in __init__
DeepEval.__init__(self, model_file, default_tf_graph = default_tf_graph)
File "/opt/miniconda3/lib/python3.8/site-packages/deepmd/DeepEval.py", line 18, in __init__
self.graph = self._load_graph (model_file, prefix = load_prefix, default_tf_graph = default_tf_graph)
File "/opt/miniconda3/lib/python3.8/site-packages/deepmd/DeepEval.py", line 47, in _load_graph
tf.import_graph_def(
File "/opt/miniconda3/lib/python3.8/site-packages/tensorflow/python/util/deprecation.py", line 507, in new_func
return func(*args, **kwargs)
File "/opt/miniconda3/lib/python3.8/site-packages/tensorflow/python/framework/importer.py", line 400, in import_graph_def
return _import_graph_def_internal(
File "/opt/miniconda3/lib/python3.8/site-packages/tensorflow/python/framework/importer.py", line 501, in _import_graph_def_internal
raise ValueError(str(e))
ValueError: NodeDef mentions attr 'T' not in Op<name=DescrptSeA; signature=coord:double, type:int32, natoms:int32, box:double, mesh:int32, davg:double, dstd:double -> descrpt:double, descrpt_deriv:double, rij:double, nlist:int32; attr=rcut_a:float; attr=rcut_r:float; attr=rcut_r_smth:float; attr=sel_a:list(int); attr=sel_r:list(int)>; NodeDef: {{node load/DescrptSeA}}. (Check whether your GraphDef-interpreting binaryis up to date with your GraphDef-generating binary.).
```

**Reply by njzjz** [reply]
Could you post all your installation scripts? I'm going to see what happened here.

**Reply by marian-code** [reply]
Yes of course. I have attached all the important files in zip attachement ([files.zip](https://github.com/deepmodeling/deepmd-kit/files/6755696/files.zip)), except the training files as those are too big.

I have a mininonda instalation. See `base.yml` for exported base environment, although I don't know if this is really important.
The environment I am using for training and inference is exported in file `dpmd_gpu.yml`. I have also included one trained potential file in zip which is throwing an error for me.

Here are all the steps needed to reproduce the error:

```bash
conda env create -f dpmd_gpu.yml
conda activate dpmd_gpu
dp train ge_se_a.yaml  # the training file also in zip attachement
dp freeze -d . -o ge_all.pb
```

The number of batches in the training file is unrealistically small but this is just a minimal example, the problem persists if the number of batches is raised to reasonable numbers e.g. 10^6

Then to get the error you just have to run this snippet in python:

```python
from deepmd.calculator import DP
calc = DP(model="ge_all_aa.pb") # in attachement
```
And you will get the error

Thank you and please let me know if you need any other details.

**Reply by njzjz** [reply]
You didn't attach `ge_se_a.yaml`, so I use `water_se_a.yaml` for test. I cannot reproduce the error.

I also use your `ge_all_aa.pb` and it works well. Your model is good.

I notice your `base.yml` uses v1.2.2 but `dpmd_gpu.yml` uses v1.3. Please check it.

**Reply by njzjz** [reply]
```sh
> grep deepmd *.yml
base.yml:  - deepmd-kit=1.2.2=py38_0_cuda10.1_gpu
base.yml:  - libdeepmd=1.2.2=0_cuda10.1_gpu
dpmd_gpu.yml:  - deepmd-kit=1.3.3=py38_1_cuda10.1_gpu
dpmd_gpu.yml:  - libdeepmd=1.3.3=0_cuda10.1_gpu
```

**Reply by marian-code** [reply]
Sorry about the yaml file I have somehow forgotten to pack it. Anyway the problem was solved by updating deepmd in base environment as you suggested. Thanks for noticing.

This surprised me. Something must have gone wrong with my conda installation. Apparently when training in `dpmd_gpu` env the model was trained correctly with `deepmd v1.3.3` but when trying to calculate with the ase interface in the same environment it seems that `deepmd v.1.2.2` was somehow loaded. I do not understand why.
